src/execution.ts

```diff
@@ -212,87 +220,106 @@ export function createBoundQuery(
   compilationContext: CompilationContext,
   document: DocumentNode,
   func: (...args: any[]) => any,
-  getVariableValues: (inputs: { [key: string]: any }) => CoercedVariableValues
+  getVariableValues: (inputs: { [key: string]: any }) => CoercedVariableValues,
+  operationName?: string
 ) {
   const {
     operation: { operation }
   } = compilationContext;
   const trimmer = createNullTrimmer(compilationContext);
   const resolvers = getFunctionResolvers(compilationContext);
-  return (
-    rootValue: any,
-    context: any,
-    variables: Maybe<{ [key: string]: any }>
-  ): Promise<ExecutionResult> | ExecutionResult => {
-    // this can be shared across in a batch request
-    const { errors, coerced } = getVariableValues(variables || {});
-
-    // Return early errors if variable coercing failed.
-    if (errors) {
-      return { errors };
-    }
+  const fnName = operationName ? operationName : "query";
+
+  /* tslint:disable */
+  /**
+   * In-order to assign a debuggable name to the bound query function,
+   * we create an intermediate object with a method named as the
+   * intended function name. This is because Function.prototype.name
+   * is not writeable.
+   *
+   * http://www.ecma-international.org/ecma-262/6.0/#sec-method-definitions-runtime-semantics-propertydefinitionevaluation
+   *
+   * section: 14.3.9.3 - calls SetFunctionName
+   */
+  /* tslint:enable */
+  const ret = {
```

Comment:
nit: I would use a more meaningful name such as `intermediateObject`. `ret` doesn't tells much about the intended purpose

Reply:
`ret` is a common variable name to indicate value used for **`ret`urn** . The comment above explains other things of using the object.